### DISCUSSION TITLE
Query menuItems by array of locations

### DISCUSSION THREAD

**Discussion by dimitryvolos** [wp-graphql/wp-graphql#2731] [Q&A]
Hi Guys,

I have multiple menus set up (WordPress), each with a custom location assigned.

I can query 'menuItems' by a single location, for example:

**menuItems(where: { location: MENU_10 }) {
        nodes {
            key: id
            url
            title
        }
    }**

Is there a way to retrieve items for multiple menus (header menu and footer menu) in a single query? Is there a way to use a single instance of 'menuItems' and use an array to specify the 'location' filter? For example:

**menuItems(where: { locations: [MENU_10, MENU_11] })**  or **menuItems(where: [{ location: MENU_10 }, { location: MENU_11 }])** 

Or should I be using a different query type?

Many thanks for your help.
D

**Comment by justlevine** [comment]
@dimitryvolos you should be able to use `menus{}` and get the `menuItems` as a child field on each menu object

**Comment by dimitryvolos** [comment]
@justlevine thank you for your reply.

I have tested `menus{}` but unfortunately, once you drill down to menuItems it still only allows you to select a single location. For example:

```
query Menus {
  menus(where: {location: MENU_10}) {
    nodes {
      menuItems {
        nodes {
          url
          title
        }
      }
      name
    }
  }
}
```
If I do not add `[location](where: {location: MENU_10})` I will receive all menus and will have to extract the data from the array manually.

Any other recommendations?

Regards,
D
